frontend/packages/ceph-storage-plugin/src/components/modals/volume-snapshot-modal/volume-snapshot-modal.tsx

```diff
@@ -29,17 +36,130 @@ export type VolumeSnapshotModalProps = {
     `${getName(resource) || 'pvc'}-snapshot-1`,
   );
+  const [scheduleLabel, setScheduleLabel] = React.useState('');
   const snapshotTypes = {
-    single: 'Single Snapshot',
+    SingleSnapshot: 'SingleSnapshot',
+    CronJob: 'CronJob',
   };
+  const selectedKey = snapshotTypes.SingleSnapshot;
+  const [scheduleType, setScheduleType] = React.useState('');
 
+  const setSchedulerHandler = (value) => {
+    setScheduleType(snapshotTypes[value]);
+  };
+
+  const getCronElements = () => {
+    const numberOfSnapshot = {
+      '3': '3',
+      '5': '5',
+      '7': '7',
+    };
+    const initialCountKey = numberOfSnapshot['3'];
+    const setSchedulerCountHandler = (value) => {
+      setSnapshotCount(parseInt(value, 10));
+    };
+    return (
+      <div>
+        <FormGroup className="ceph-volume-snapshot-modal--input" fieldId="snapshot-modal__schedule">
+          <label className="pf-c-form__label" htmlFor="snapshot-modal__schedule">
+            <span className="pf-c-form__label-text">Label</span>
+            <a
+              className="ceph-volume-snapshot-modal--link"
+              href="https://en.wikipedia.org/wiki/Cron#Overview"
+            >
+              Whats this?
+            </a>
+          </label>
+          <TextInput
+            isRequired
+            type="text"
+            name="snapshot-modal__schedule"
+            aria-label="snapshot-modal__schedule"
+            value={scheduleLabel}
+            onChange={(value) => setScheduleLabel(value)}
+            className="ceph-volume-snapshot-modal--label"
+            placeholder="**5***"
+          />
+          <span className="help-block">* Scheduled for UTC+3</span>
+        </FormGroup>
+        <FormGroup
+          className="ceph-volume-snapshot-modal--input"
+          label="Keep"
+          fieldId="snapshot-modal__count"
+        >
+          <div className="ceph-volume-snapshot-modal--input_keep">
+            <Dropdown
+              items={numberOfSnapshot}
+              selectedKey={initialCountKey}
+              onChange={(value) => setSchedulerCountHandler(value)}
+              className="ceph-volume-snapshot-modal--count"
+            />
+            <span className="ceph-volume-snapshot-modal--span_keep">last snapshots</span>
+          </div>
+        </FormGroup>
+        <span className="help-block">*Older snapshots will be deleted automatically</span>
+      </div>
+    );
+  };
+
+  const makeSchedule = (scheduleTemplate: K8sResourceKind) => {
+    handlePromise(k8sCreate(SnapshotScheduleModel, scheduleTemplate))
+      .then(close)
+      .catch((error) => {
+        throw error;
+      });
+  };
   const submit = (event: React.FormEvent<EventTarget>) => {
     event.preventDefault();
     const ns = getNamespace(resource);
     const pvcName = getName(resource);
+    if (scheduleType === 'CronJob') {
+      const pvcLabel = Math.random()
+        .toString(36)
+        .slice(2);
```

Comment:
Can you explain the use of `36` and `2`

Reply:
radix 36 should convert the random value to an alpha-numeric string. 0-9 a-z